spec/collator.html

```diff
@@ -25,40 +22,18 @@ <h1>Intl.Collator ( [ _locales_ [ , _options_ ] ] )</h1>
-        1. Set _opt_.[[localeMatcher]] to _matcher_.
-        1. Let _collation_ be ? GetOption(_options_, *"collation"*, ~string~, ~empty~, *undefined*).
-        1. If _collation_ is not *undefined*, then
-          1. If _collation_ cannot be matched by the <code>type</code> Unicode locale nonterminal, throw a *RangeError* exception.
-        1. Set _opt_.[[co]] to _collation_.
-        1. Let _numeric_ be ? GetOption(_options_, *"numeric"*, ~boolean~, ~empty~, *undefined*).
-        1. If _numeric_ is not *undefined*, then
-          1. Set _numeric_ to ! ToString(_numeric_).
-        1. Set _opt_.[[kn]] to _numeric_.
-        1. Let _caseFirst_ be ? GetOption(_options_, *"caseFirst"*, ~string~, « *"upper"*, *"lower"*, *"false"* », *undefined*).
-        1. Set _opt_.[[kf]] to _caseFirst_.
-        1. Let _relevantExtensionKeys_ be %Intl.Collator%.[[RelevantExtensionKeys]].
-        1. Let _r_ be ResolveLocale(%Intl.Collator%.[[AvailableLocales]], _requestedLocales_, _opt_, _relevantExtensionKeys_, _localeData_).
+        1. Let _optionsResolution_ be ? ResolveOptions(%Intl.Collator%, _localeData_, CreateArrayFromList(_requestedLocales_), _options_).
+        1. Let _r_ be _optionsResolution_.[[ResolvedLocale]].
         1. Set _collator_.[[Locale]] to _r_.[[Locale]].
-        1. Set _collation_ to _r_.[[co]].
-        1. If _collation_ is *null*, set _collation_ to *"default"*.
+        1. If _r_.[[co]] is *null*, let _collation_ be *"default"*. Otherwise, let _collation_ be _r_.[[co]].
         1. Set _collator_.[[Collation]] to _collation_.
-        1. If _relevantExtensionKeys_ contains *"kn"*, then
-          1. Set _collator_.[[Numeric]] to SameValue(_r_.[[kn]], *"true"*).
-        1. If _relevantExtensionKeys_ contains *"kf"*, then
-          1. Set _collator_.[[CaseFirst]] to _r_.[[kf]].
+        1. Set _collator_.[[Numeric]] to SameValue(_r_.[[kn]], *"true"*).
+        1. Set _collator_.[[CaseFirst]] to _r_.[[kf]].
         1. Let _resolvedLocaleData_ be _r_.[[LocaleData]].
-        1. Let _sensitivity_ be ? GetOption(_options_, *"sensitivity"*, ~string~, « *"base"*, *"accent"*, *"case"*, *"variant"* », *undefined*).
-        1. If _sensitivity_ is *undefined*, then
-          1. If _usage_ is *"sort"*, then
-            1. Set _sensitivity_ to *"variant"*.
-          1. Else,
-            1. Set _sensitivity_ to _resolvedLocaleData_.[[sensitivity]].
-        1. Set _collator_.[[Sensitivity]] to _sensitivity_.
+        1. If _usage_ is *"sort"*, let _defaultSensitivity_ be *"variant"*. Otherwise, let _defaultSensitivity_ be _resolvedLocaleData_.[[sensitivity]].
+        1. Set _collator_.[[Sensitivity]] to ? GetOption(_options_, *"sensitivity"*, ~string~, « *"base"*, *"accent"*, *"case"*, *"variant"* », _defaultSensitivity_).
         1. Let _defaultIgnorePunctuation_ be _resolvedLocaleData_.[[ignorePunctuation]].
```

Comment:
Can this line be collapsed into the next one as well?

Reply:
Yes, but I think the _defaultIgnorePunctuation_ name is worth keeping (especially as an analog to _defaultSensitivity_).

Reply:
Yeah, that's fair enough. Alright then I'll take a quick skim yet again today given the size of this patch to make sure everything looks alright once again and hit merge.